### PR TITLE
Adds missing form data arguments in hooks

### PR DESCRIPTION
**Pull request by tomas862**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.6.x
| Description?  | As a module developer when I want to get form data withing identifiable object hook I need to get it from **Request** object - things would be way easier if we can just retrieve data from hook argument directly
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | https://github.com/PrestaShop/PrestaShop/issues/13944
| How to test?  | This PR is not in testable state - the only test will be achieved when this update will be used directly in module ( I have sample module already using this place so I'll modify some code there once this is merged)

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/14008)

### CODE REVIEW

**Review by PierreRambaud — DISMISSED** [review]

File: src/Core/Form/IdentifiableObject/Handler/FormHandler.php [hunk 1]
```diff
@@ -144,6 +144,7 @@ private function handleFormUpdate(FormInterface $form, $id)
 
         $this->hookDispatcher->dispatchWithParameters('actionAfterUpdate' . Container::camelize($form->getName()) . 'FormHandler', [
             'id' => $id,
+            'form_data' => $data,
```
Comment (PierreRambaud) [inline src/Core/Form/IdentifiableObject/Handler/FormHandler.php]: If this data is used by many modules, should it be passed by reference in order to get changes between modules? :thinking:

Comment (matks) [reply src/Core/Form/IdentifiableObject/Handler/FormHandler.php]: The usecase @PierreRambaud think of is:
- module 1 that is hooked here modifies `form_data`
- module 2 that is hooked here too wants to modify `form_data` but it gets the initial `form_data`, without the modifications done by module 1

Comment (matks) [reply src/Core/Form/IdentifiableObject/Handler/FormHandler.php]: Requested change applied in https://github.com/PrestaShop/PrestaShop/pull/14008/commits/3d4336b5338b87f6c5888ea1cec48b6baac2786d

**Review by PierreRambaud — APPROVED** [review]

**Comment by matks** [comment]
Thank you @tomas862

**Pull request merged**